index.html

```diff
@@ -432,50 +444,62 @@
       <div id="nav-popout"> 
 
       </div>
-      
+
       <div class="cardContainer" id="cardContainer">
 
       </div>
 
+      <div id="metamaskWarning">
+        <img id="metamaskImg" src="./metamask.png">
+        <h3><a href="https://metamask.io/" target="_blank">Metamask Extension</a> required to use Exokit Store.</h3>
+      </div>
+
       <div class=multibutton>
         <a class="button first last" id="enter-xr-button" style="display: none;">Enter XR</a>
         <a class="button first last" id="no-xr-button" disabled style="display: none;">No XR</a>
       </div>
     </div>
     <script>
 (async () => {
+
+  const metamaskWarning = document.getElementById("metamaskWarning");
+
   const contract = await new Promise((accept, reject) => {
     window.addEventListener('load', async () => {
+
+      const accessContract = async (web3) => {
+        window.web3 = web3;
+        const ress = await Promise.all([
+          fetch('abi.json'),
+          fetch('address.json'),
+        ]);
+        const [jsonInterface, address] = await Promise.all(ress.map(res => res.json()));
+        return web3.eth.contract(jsonInterface).at(address);
+      }
+
       // Modern dapp browsers...
       if (window.ethereum) {
-          window.web3 = new window.top.Web3(ethereum);
           try {
-              // Request account access if needed
-              await window.ethereum.enable();
-              // Acccounts now exposed
-              // web3.eth.sendTransaction({/* ... */});
+            // Request account access if needed
+            await window.ethereum.enable();
+            const contract = await accessContract(new window.top.Web3(ethereum))
+            accept(contract);
+
           } catch (error) {
-              // User denied account access...
+              metamaskWarning.style.display = "block";
+              console.warn("User denied access to Metamask account.")
```

Comment:
The promise will hang here. There will be no accept _or_ reject.

I think there should only be a single `accept`/reject at the end of this stack of code, not little accepts and rejects sprinkled everywhere. Otherwise it's easy for hangs like this to creep in.